### PR TITLE
Add storybook-router (replaces #1000)

### PR DESCRIPTION
**Pull request by GAntoine**
Credit to @pascalduez for the work

### CODE REVIEW

File: definitions/npm/storybook-router_v0.x.x/flow_v0.25.x-/storybook-router_v0.x.x.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  hash?: string,
+  state?: any
+};
+
+type GetUserConfirmation = (
+  message: string,
+  callback: (confirmed: boolean) => void
+) => void;
+
+type Renderable = React$Element<*>;
```
Comment (sibelius) [inline definitions/npm/storybook-router_v0.x.x/flow_v0.25.x-/storybook-router_v0.x.x.js]: can't we import this from React ?

import type { Element } from 'react'?

or is it better to use private def?

Comment (GAntoine) [reply definitions/npm/storybook-router_v0.x.x/flow_v0.25.x-/storybook-router_v0.x.x.js]: Yeah I think that's safe. Would it be `type Renderable = Element<*>` or `type Renderable = Element`?

File: definitions/npm/storybook-router_v0.x.x/flow_v0.25.x-/storybook-router_v0.x.x.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import type { Element } from "react";
```
Comment (pascalduez) [inline definitions/npm/storybook-router_v0.x.x/flow_v0.25.x-/storybook-router_v0.x.x.js]: Isn't ` import type { Element } from "react"; ` only available from Flow >= 0.53?
In such case Element would resolve `any` for all other lower versions.

Comment (GAntoine) [reply definitions/npm/storybook-router_v0.x.x/flow_v0.25.x-/storybook-router_v0.x.x.js]: Correct, I figured it's not a huge deal if it does resolve to `any`. If you're keen to fix it, a PR would be much appreciated!

Comment (pascalduez) [reply definitions/npm/storybook-router_v0.x.x/flow_v0.25.x-/storybook-router_v0.x.x.js]: Your right, that's fine as is I guess. ;)

Comment (pascalduez) [reply definitions/npm/storybook-router_v0.x.x/flow_v0.25.x-/storybook-router_v0.x.x.js]: Thanks for all the hard work!

**Pull request merged**